src/Microsoft.Diagnostics.Runtime/src/DataTargets/DataTarget.cs

```diff
@@ -239,86 +237,102 @@ public IEnumerable<ModuleInfo> EnumerateModules()
-        /// <returns>A <see cref="DataTarget"/> for the given dump file.</returns>
-        public static DataTarget LoadDump(string filePath, CacheOptions? cacheOptions = null)
+        /// <param name="displayName">The name of this DataTarget, might be used in exceptions.</param>
+        /// <param name="stream">The stream that should be used.</param>
+        /// <param name="cacheOptions">The caching options to use. (Only used for FileStreams)</param>
+        /// <param name="leaveOpen">True whenever the given stream should be leaved open when the DataTarget is disposed.</param>
+        /// <returns>A <see cref="DataTarget"/> for the given dump.</returns>
+        public static DataTarget LoadDump(string displayName, Stream stream, CacheOptions? cacheOptions = null, bool leaveOpen = false)
         {
-            if (filePath is null)
-                throw new ArgumentNullException(nameof(filePath));
-            else if (!File.Exists(filePath))
-                throw new FileNotFoundException($"Could not open dump file '{filePath}'.", filePath);
-
-            cacheOptions ??= new CacheOptions();
-
-            (FileStream stream, DumpFileFormat format) = OpenDump(filePath);
             try
             {
+                if (displayName is null)
+                    throw new ArgumentNullException(nameof(displayName));
+                if (stream is null)
+                    throw new ArgumentNullException(nameof(stream));
+
+                cacheOptions ??= new CacheOptions();
+
+                DumpFileFormat format = ReadFileFormat(stream);
+
 #pragma warning disable CA2000 // Dispose objects before losing scope
 
                 IDataReader reader = format switch
                 {
-                    DumpFileFormat.Minidump => new MinidumpReader(filePath, stream, cacheOptions),
-                    DumpFileFormat.ElfCoredump => new CoredumpReader(filePath, stream),
+                    DumpFileFormat.Minidump => new MinidumpReader(displayName, stream, cacheOptions, leaveOpen),
+                    DumpFileFormat.ElfCoredump => new CoredumpReader(displayName, stream, leaveOpen),
 
                     // USERDU64 dumps are the "old" style of dumpfile.  This file format is very old and shouldn't be
                     // used.  However, IDebugClient::WriteDumpFile(,DEBUG_DUMP_DEFAULT) still generates this format
                     // (at least with the Win10 system32\dbgeng.dll), so we will support this for now.
-                    DumpFileFormat.Userdump64 => new DbgEngDataReader(filePath, stream),
-                    DumpFileFormat.CompressedArchive => throw new InvalidDataException($"File '{filePath}' is a compressed archived instead of a dump file."),
-                    _ => throw new InvalidDataException($"File '{filePath}' is in an unknown or unsupported file format."),
+                    DumpFileFormat.Userdump64 => new DbgEngDataReader(displayName, stream, leaveOpen),
```

Comment:
I've just noticed that Userdump64 cannot be created by a stream :/

Reply:
It's fine, just throw a NotSupportedException for that case.